R/utils.R

```diff
@@ -259,84 +259,333 @@ rowcol_to_dim <- function(row, col) {
   # we will always return something like "A1"
   stringi::stri_join(min_col, min_row)
 }
-#' @rdname dims_helper
-#' @param ... construct dims arguments, from rows/cols vectors or objects that can be coerced to data frame
+#' Helper to specify the `dims` argument.
+#'
+#' @description
+#' `wb_dims()` can be used to help provide the `dims` argument, in the `wb_add_*` functions.
+#' It returns a Excel range (i.e. "A1:B1") or a start like "A2".
+#' It can be very useful as you can specify many parameters that interact together
+#' In general, you must provide named arguments. `wb_dims()` will only accept unnamed arguments
+#' if it is `rows`, `cols`, for example `wb_dims(1:4, 1:2)`, that will return "A1:B4".
+#'
+#' `wb_dims()` can also be used with an object (a `data.frame` or a `matrix` for example.)
+#' All parameters are numeric unless stated otherwise.
+#' # Using `wb_dims()` alone
+#'
+#' * `rows` / `cols` (if you want to specify a single one, use `start_row` / `start_col`)
+#' * `start_row`
+#' * `start_col`
+#'
+#' # Using `wb_dims()` with an object
+#'
+#' * `x` An object (typically a `matrix` or a `data.frame`)
+#' * `start_row` the starting row of `x` (The `dims` returned will be )
+#' * `start_col` the starting column: a single integer, or an Excel column identifier "A", "B" etc.
+#' * `rows` (Which rows? (not fully supported yet
+#' * `cols` a range of column, or one of the column names of `x` (length 1 only accepted)
```

Comment:
In this case `dims_to_dataframe()` identifies the `;`, but still creates a square matrix and not a list of matrices. And in addition, our styles functions do not handle these. I didn't want to add to much overhead in the functions for conditions the user could simply run in a loop. But of course this would be possible to handle, just ... a new feature.

Maybe: 

```R
foo <- function(dims, bar = wb_color()) {
  if (is.list(dims)) {
     for (i in seq_along(dims)){
        self$foo(dims[[i]], bar = bar)
  }
 ...
}

Reply:
Indeed many ranges seems out of scope here. 
But maybe eventually, they could be supported

Reply:
I just wanted to clarify that this is currently not possible and nothing I have worked on or plan to do in the upcoming weeks.